src/JustSaying/AwsTools/MessageHandling/SnsTopicByName.cs

```diff
@@ -66,14 +68,34 @@ public async Task EnsurePolicyIsUpdatedAsync(IReadOnlyCollection<string> additio
             }
         }
 
+        public async Task ApplyTagsAsync()
+        {
+            if (!Tags.Any())
+            {
+                return;
+            }
+
+            Tag CreateTag(KeyValuePair<string, string> tag) => new() { Key = tag.Key, Value = tag.Value };
+
+            var tagRequest = new TagResourceRequest
+            {
+                ResourceArn = Arn,
+                Tags = Tags.Select(CreateTag).ToList()
+            };
+
+            await Client.TagResourceAsync(tagRequest).ConfigureAwait(false);
+
+            _logger.LogInformation("Added {TagCount} tags to topic {TopicName}", tagRequest.Tags.Count, TopicName);
```

Comment:
Maybe we should word this differently? If it had 3 tags before and you set a different 3, you'd still have 3 tags as the old 3 would have been removed rather than _adding_ 3 extra?

Reply:
I don’t believe old tags are removed. You’d just replace any that had the same key

Reply:
Ah ok, I thought it was more of a "tags are these and these only" operation than additive.